blockbase/inc/customizer/wp-customize-fonts.php

```diff
@@ -3,552 +3,177 @@
+	$body_font_slug    = '';
+	foreach ( $font_settings['custom'] as $font_setting ) {
+		if ( strpos( $font_setting['slug'], 'heading' ) !== false ) {
+			$heading_font_slug = $font_setting['fontSlug'];
 		}
-	}
-
-	function enqueue_google_fonts() {
-		wp_enqueue_style( 'blockbase-google-fonts', $this->google_fonts_url(), array(), null );
-	}
 
-	function create_customization_style_element( $wp_customize ) {
-		if ( $this->font_settings ) {
-			wp_enqueue_style( 'global-styles-fonts-customizations', ' ', array( 'global-styles' ) ); // This needs to load after global_styles, hence the dependency
-			$css  = 'body {';
-			$css .= '--wp--preset--font-family--body-font:' . $this->font_settings['body'] . ';';
-			$css .= '--wp--preset--font-family--heading-font:' . $this->font_settings['heading'] . ';';
-			$css .= '}';
-			wp_add_inline_style( 'global-styles-fonts-customizations', $css );
+		if ( strpos( $font_setting['slug'], 'body' ) !== false ) {
+			$body_font_slug = $font_setting['fontSlug'];
 		}
 	}
 
-	function update_font_settings( $wp_customize ) {
-		$body_setting    = $wp_customize->get_setting( $this->section_key . 'body' );
-		$heading_setting = $wp_customize->get_setting( $this->section_key . 'heading' );
-		if ( $body_setting && $heading_setting ) {
-			$body_value = $body_setting->post_value();
-			if ( $body_value ) {
-				$body_font_setting           = $this->fonts[ $body_value ];
-				$this->font_settings['body'] = $body_font_setting['fontFamily'];
-			}
+	// Get the user's global styles CPT id
+	$user_custom_post_type_id       = WP_Theme_JSON_Resolver::get_user_global_styles_post_id();
+	$global_styles_controller       = new WP_REST_Global_Styles_Controller();
+	$global_styles                  = fetch_global_styles( $user_custom_post_type_id, $global_styles_controller );
+	$blockbase_legacy_font_settings = $global_styles->data['settings']['typography']['fontFamilies']['custom'];
 
-			$heading_value = $heading_setting->post_value();
-			if ( $heading_value ) {
-				$heading_font_setting           = $this->fonts[ $heading_value ];
-				$this->font_settings['heading'] = $heading_font_setting['fontFamily'];
-			}
-		}
-	}
-
-	function google_fonts_url() {
-		$font_families = array();
-		foreach ( $this->fonts as $font ) {
-			if ( ! empty( $font['google'] ) ) {
-				$font_families[] = $font['google'];
-			}
-		}
-		$font_families[] = 'display=swap';
+	// converts data to array (in some cases settings and styles are objects insted of arrays)
+	$new_settings = (array) $global_styles->data['settings'];
+	$new_styles   = (array) $global_styles->data['styles'];
 
-		// Make a single request for the theme fonts.
-		return esc_url_raw( 'https://fonts.googleapis.com/css2?' . implode( '&', $font_families ) );
+	// Set new typography settings
+	if ( isset( $new_settings['typography']['fontFamilies'] ) ) {
+		unset( $new_settings['typography']['fontFamilies'] ); // TODO: Reconsider the depth of property we're unsetting
```

Comment:
I believe we tested this and it needed to be `['typography']['fontFamilies']`, as `['typography']['fontFamilies']['custom']` didn't work.

Reply:
> I believe we tested this and it needed to be ['typography']['fontFamilies'], as ['typography']['fontFamilies']['custom'] didn't work.

Yes! You're right. I think it's a case of us forgotting to remove the inline comment

Reply:
Removed in https://github.com/Automattic/themes/pull/6010/commits/1782be8707c829cd81d84732317c703abfd57d7c